### DISCUSSION TITLE
Dagster-ecs unable to find dagster_postgres

### DISCUSSION THREAD

**Discussion by hmalikunlearn** [dagster-io/dagster#18455] [Q&A] [answered]
Hello, 

I am running a dagster on an ecs cluster and during the job run I am getting the following error 
```
ModuleNotFoundError: No module named 'dagster_postgres
```

My docker file looks like this 

```
# Dagster
FROM python:3.11.3-slim as dagster

RUN apt-get update && apt-get upgrade -yqq
RUN apt-get install git -y
ENV DAGSTER_HOME=/opt/dagster/dagster_home/
RUN mkdir -p /opt/dagster/dagster_home/
WORKDIR $DAGSTER_HOME
COPY dagster.yaml workspace.yaml /opt/dagster/dagster_home/
RUN git clone https://github.com/dagster-io/dagster.git

# Install:
# - dagster so we can run `dagster-daemon run`
# - dagster-aws so we can use EcsRunLauncher
# - dagster-postgres so we can use PostgresEventStorage,
#   PostgresRunStorage, and PostgresScheduleStorage
COPY requirements-dagster.txt /opt/dagster/dagster_home/
RUN pip install -r requirements-dagster.txt

# webserver
FROM dagster as webserver
COPY requirements-webserver.txt /opt/dagster/dagster_home/
RUN pip install -r requirements-webserver.txt
```

And my requirements-webserver.txt is this 

```
dagster
dagster-aws
dagster-postgres
``` 

Previously we were using dagit for our ecs workload but now after upgrading to using the webserver we are now running into this error. I am not sure why this error is occurring since during the docker build process I am able to see that the module is installed correctly. Can someone please help with this ?

**Comment by gibsondan** [comment]
Can you include the full stack trace of the error that you're seeing? That will help determine exactly where the problem is (for example, it could be missing in your webserver image, or it could be missing in the image where your code runs when the run i slaunched).

A good way to check if dagster-postgres is actually installed is to open a shell using the Docker image and run:

`pip freeze | grep dagster`

That will show you if dagster-postgres is actually installed there.

You'll want dagster-postgres to be installed in any images that you are using that are running Dagster code.

**Reply by hmalikunlearn** [reply]
Thank you for the suggestion, i've looked at it a bit further and it seems that tasks are using an older image version compared to the user_code which is at a higher version that includes the dagster_postgres package. 

Is there a way to ensure that this is always consistent since im seeing that there are a few different task definitions that are still showing as active but are actually older.

**Reply by gibsondan** [reply]
Can you share more details? Which tasks are using the older image?

**Reply by hmalikunlearn** [reply]
Our user_code is running version an image tagged 'test-modules'. When i run a job inside of dagster a task is created in ECS and when i look at the image the task is running it is using the user_code image tagged 'v2023.11.2'. This older image does not have the module installed in it and I am not sure why this image is being picked up instead of the deployed version of user_code

**Reply by gibsondan** [reply, marked as answer]
The way that our ECS example specifies what image to use in runs that are launched is described here: https://github.com/dagster-io/dagster/blob/1.5.10/examples/deploy_docker/docker-compose.yml#L16-L35 - it specifies an environment variable in the user code task that is then picked up when the run launched.

To provide more specific guidance we'll need more information about your deployment setup / a way to reproduce the problem that you're seeing.

**Reply by hmalikunlearn** [reply]
Thanks you @gibsondan that resolved the issue :) I calling the incorrect version there
